e2e/cypress/integration/custom_status/custom_status_spec.js

```diff
@@ -0,0 +1,557 @@
+            cy.get('.status-dropdown-menu .custom_status__row span.emoticon').invoke('attr', 'data-emoticon').should('contain', customStatus.emoji);
+
+            // * Clear button should be visible in the status dropdown
+            cy.get('.status-dropdown-menu #custom_status__clear').should('exist');
+        });
+
+        it('MM-T3846_6 should clear the custom status text when clear button is clicked', () => {
+            // # Click on the clear button in the status dropdown
+            cy.get('.status-dropdown-menu #custom_status__clear').click();
+
+            // * Custom status text should be removed and "Set a Custom Status" should be displayed in the status dropdown
+            cy.get('.status-dropdown-menu .custom_status__row').should('have.text', 'Set a Custom Status');
+        });
+
+        it('MM-T3846_7 should show previosly set status in the first position in Recents list', () => {
+            // # Click on the "Set a Custom Status" option in the status dropdown
+            cy.get('.status-dropdown-menu li#status-menu-custom-status').click();
+
+            // * Custom status modal should open
+            cy.get('#custom_status_modal').should('exist');
+
+            // * Previously set status should be first in the recents list along with the correct emoji
+            cy.get('#custom_status_modal .statusSuggestion__row').first().find('.statusSuggestion__text').should('have.text', customStatus.text);
+            cy.get('#custom_status_modal .statusSuggestion__row').first().find('span.emoticon').invoke('attr', 'data-emoticon').should('contain', customStatus.emoji);
+        });
+
+        it('MM-T3846_8 should set the same status again when clicked on the Set status', () => {
+            // # Select the first suggestion from the list and set the status
+            cy.get('#custom_status_modal .statusSuggestion__row').first().click();
+            cy.get('#custom_status_modal .GenericModal__button.confirm').click();
+
+            // * Custom status modal should be closed
+            cy.get('#custom_status_modal').should('not.exist');
+
+            // * Correct custom status emoji should be displayed in the sidebar header
+            cy.get('#headerInfoContent span.emoticon').invoke('attr', 'data-emoticon').should('contain', customStatus.emoji);
+        });
+
+        it('MM-T3846_9 should clear the status when clicked on Clear status button', () => {
+            openCustomStatusModal();
+
+            // # Click on the Clear status button
+            cy.get('#custom_status_modal').findByText('Clear Status').click();
+
+            // # Open status dropdown
+            cy.get('.MenuWrapper .status-wrapper').click();
+
+            // * Custom status text should not be displayed in the status dropdown
+            cy.get('.status-dropdown-menu .custom_status__row').should('not.have.text', customStatus.text);
+        });
+    });
+
+    describe('MM-T3847 Recent statuses', () => {
```

Comment:
Please move to its own file `custom_status_4_spec.js`

```suggestion
    describe('Custom Status - Recent Statuses', () => {
```